praw/objects.py

```diff
@@ -640,15 +669,113 @@ def _update_submission(self, submission):
                 reply._update_submission(submission)  # pylint: disable=W0212
 
     @property
+    def default_contextlink(self):
+        """Return the default link to the comment's context."""
+        context_suffix = self.id + "?context=3"
+        return urljoin(self.submission_permalink, context_suffix)
+
+    def contextlink(self, context_number=3):
+        """Return a link to the comment's context.
+
+        :param context_number: The amount of context in the tree. Must be
+        a integer. Defaults to 3 as on reddit.
+
+        """
+        if not isinstance(context_number, int):
```

Comment:
I don't think we need to do that check. The few places explicit `TypeErrors` should occur is when checking a parameter than intentionally can be more than one type with different behavior per type, and when checking that optional parameters are only used in conjunction with other parameters.